### ISSUE TITLE
epic: Complete InCluster Networking

### ISSUE THREAD

**Issue by davidfestal** [kcp-dev/contrib-tmc#95]
### TL;DR

Extend the limited In Cluster Network support provided in PR kcp-dev/kcp#1708, in order to add:
- Support for Syncing from multiple User Workspaces from given Syncer
- NetworkPolicy-based protection of the accesses to the CoreDNS services installed for each user orkspace 

### Progress tracking <a href="#progress-tracking" id="progress-tracking">#</a>

EPIC detailed issues and overall progress can be tracked with the following [project view](https://github.com/orgs/kcp-dev/projects/1/views/41)

### Work Items

- [x] kcp-dev/kcp#1987
- [x] kcp-dev/kcp#1988
- [ ] kcp-dev/contrib-tmc#80
- [ ] kcp-dev/kcp#1531
